bazel/external/quiche.patch

```diff
@@ -0,0 +1,51 @@
+diff --git quic/core/quic_connection.cc quic/core/quic_connection.cc
+index 5f645bd7..85c0d411 100644
+--- quic/core/quic_connection.cc
++++ quic/core/quic_connection.cc
+@@ -4457,7 +4457,7 @@ bool QuicConnection::FlushCoalescedPacket() {
+   if (!buffered_packets_.empty() || HandleWriteBlocked()) {
+     QUIC_DVLOG(1) << ENDPOINT
+                   << "Buffering coalesced packet of len: " << length;
+-    buffered_packets_.emplace_back(buffer, length,
++    buffered_packets_.emplace_back(buffer, static_cast<QuicPacketLength>(length),
+                                    coalesced_packet_.self_address(),
+                                    coalesced_packet_.peer_address());
+     if (debug_visitor_ != nullptr) {
+diff --git quic/core/quic_connection.cc quic/core/quic_connection.cc
+index 5f645bd7..43155e74 100644
+--- quic/core/quic_connection.cc
++++ quic/core/quic_connection.cc
+@@ -4484,7 +4484,7 @@ bool QuicConnection::FlushCoalescedPacket() {
+     if (result.status != WRITE_STATUS_BLOCKED_DATA_BUFFERED) {
+       QUIC_DVLOG(1) << ENDPOINT
+                     << "Buffering coalesced packet of len: " << length;
+-      buffered_packets_.emplace_back(buffer, length,
++      buffered_packets_.emplace_back(buffer, static_cast<QuicPacketLength>(length),
+                                      coalesced_packet_.self_address(),
+                                      coalesced_packet_.peer_address());
+     }
+diff --git quic/core/quic_sent_packet_manager.cc quic/core/quic_sent_packet_manager.cc
+index a9204d67..e9c61327 100644
+--- quic/core/quic_sent_packet_manager.cc
++++ quic/core/quic_sent_packet_manager.cc
+@@ -766,6 +766,8 @@ QuicSentPacketManager::OnRetransmissionTimeout() {
+       ++consecutive_pto_count_;
+       pending_timer_transmission_count_ = max_probe_packets_per_pto_;
+       return PTO_MODE;
++    default:
```

Comment:
This part of the patch can be revised if there is a quiche codebase convention for aborting/panicking etc., this is related to the compilation error:

```
Execution platform: @local_config_platform//:host
C:\_eb\execroot\envoy\bazel-out\x64_windows-opt\bin\external\com_googlesource_quiche\quiche\quic\core\quic_sent_packet_manager.cc(770) : error C2220: the following warning is treated as an error
C:\_eb\execroot\envoy\bazel-out\x64_windows-opt\bin\external\com_googlesource_quiche\quiche\quic\core\quic_sent_packet_manager.cc(770) : warning C4715: 'quic::QuicSentPacketManager::OnRetransmissionTimeout': not all control paths return a value
Target //test/extensions/quic_listeners/quiche:envoy_quic_utils_test failed to build
```

Reply:
fixing it in https://github.com/envoyproxy/envoy/pull/13066